lib/src/graph/package_graph.dart

```diff
@@ -168,9 +168,10 @@ class PackageGraph {
       return new Future.error(error, _lastUnexpectedErrorTrace);
     }
 
-    // If the last build completed with an error, complete the future with it.
-    if (!_lastResult.succeeded) {
-      return new Future.error(BarbackException.aggregate(_lastResult.errors));
+    // If any build completed with an error, complete the future with it.
+    var result = new BuildResult.aggregate(_results);
```

Comment:
Discussed offline.

This change may be counter to barback's design. barback doesn't have an explicit lifecycle, and so there is no obvious point at which we can clear recorded errors. This may be why only the most recent BuildResult is considered today - we expect that the client of barback is already listening on the error stream and knows better than barback does which errors/lifespan is relevant.

I'm going to go back to the change in pub and focus on getting a test in place there.